core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -762,6 +768,12 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusContext context)
         ensureFullPathAndUpdateCache(inodePath);
       }
       FileInfo fileInfo = getFileInfoInternal(inodePath);
+      Mode.Bits accessMode = Mode.Bits.fromProto(context.getOptions().getAccessMode());
+      if (context.getOptions().hasAccessMode()
+           && (accessMode.imply(Mode.Bits.READ) || accessMode.imply(Mode.Bits.WRITE))) {
```

Comment:
We want to limit access time update to only apply to file content access. 

From user perspective, it is more useful for knowing when user actually opened the file and read data, rather than just someone listed or changed attributes, especially because the policy engine is focused on moving the actual data than the metadata.

From technical perspective, updating inode tree and journal on every `getStatus` call is just too expensive, even with all the optimization in the update logic. It effectively makes every metadata read a write operation.

Reply:
Yes it is consistent with HDFS. HDFS update access time on acquiring block locations, which is only called when a client reads a file. getFileInfo is called by client for many metadata only operations such as checking existence or running stat command. The access mode is an optional parameter for the `getFileInfo()` RPC. It is currently only set by the client when used inside the client file `open` API.